Classes/ScrnSPThompsonFire.uc

```diff
@@ -66,6 +66,14 @@ state FireLoop
 
         NextFireTime = Level.TimeSeconds - 0.000001; //fire now!
     }
+    function ModeTick(float dt)
```

Comment:
Do we need to do it every tick? Isn't it enough to do it at the `EndState()`? Please check.

Reply:
Good question, I have no idea actually, I just copied what was in the other TommyGuns:
(From ScrnThompsonIncFire.uc and ScrnThompsonDrumFire.uc)

```
// fixes double shot bug -- PooSH
state FireLoop
{
    function BeginState()
    {
        super.BeginState();

        NextFireTime = Level.TimeSeconds - 0.000001; //fire now!
    }
    function ModeTick(float dt)
    {
        if( KFWeap.MagAmmoRemaining < 1 )
        {
            DoCloseBolt(); //plays sound and sets bBoltClosed
        }
        Super.ModeTick(dt);
    }
}
```